README.md

```diff
@@ -0,0 +1,222 @@
+# function-shell
+
+This Crossplane composition function accepts commands to run in a shell and it
+returns the output to specified fields. It accepts the following paramereters:
+- `shellEnvVarsSecretRef` - referencing environment variables in a
+  Kubernetes secret. shellEnvVarsSecretRef requires a `name`, a
+`namespace` and a `key` for the secret. Inside of it, the shell
+expects a JSON structure with key value environment variables. Example:
+
+```
+{
+    "ENV_FOO": "foo value",
+    "ENV_BAR": "bar value"
+}
+```
+- 'shellEnvVars' - an array of environment variables with a <b>key</b> and <b>value</b>
+  each.
+- 'shellCommand' - a shell command line that can contain pipes and
+  redirects and calling multiple programs.
+- 'stdoutField' - the path to the field where the shell standard output
+  should be written.
+- 'stderrField' - the path to the field where the shell standard error
+  output should be written.
+
+## Practical Example: Obtain Dashboard Ids from Datadog
+
+The composition calls the `function-shell` instructing it to obtain dashboard ids
+from a Datadog account. For this, it specifies the location of a Kubernetes
+secret where the `DATADOG_API_KEY` and `DATADOG_APP_KEY` environment variable values
+are stored. The Datadog API endpoint is passed in a clear text environment
+variable. The shell command uses a `curl` to the endpoint with a header that
+contains the access credentials. The command output is piped into jq and
+filtered for the ids.
+
+The `function-shell` writes the dashboard ids to the specified output status field, and
+any output that went to stderr into the specified stderr status field.
+
+The composition is for illustration purposes only. When using the
+`function-shell` in your own compositions, you may want to patch function input
+from claim and other composition field values.
+
+Note: `function-shell` has to receive permissions in form of a rolebinding to
+read secrets and perform other actions that may be prohibited by default. Below
+is a `clusterrolebinding` that will work, but you should exercise appropriate caution when
+setting function permissions.
+
+```
+#!/bin/bash
+SA=$(kubectl -n upbound-system get sa -o name | grep function-shell | sed -e 's|serviceaccount\/|upbound-system:|g')
```

Comment:
The usage of `upbound-system` should probably be replaced with `crossplane-system` so crossplane community folks can easily use these examples, what do you think?

Reply:
Updated. Will be in next PR.